timeutil/schedule.go

```diff
@@ -215,3 +395,341 @@ func ParseSchedule(scheduleSpec string) ([]*Schedule, error) {
+	// explicitly indicated, thus the parsing will first try to handle it as
+	// a wdayset, then as timeset
+	if els := strings.Split(s, ","); len(els) > 1 {
+		events = els
+	} else {
+		events = []string{s}
+	}
+
+	// temporary schedules
+	var tempScheds []*Schedule
+	// currently collected schedules
+	var scheds []*Schedule
+	// indicates that any further events must be time events
+	var expectTime bool
+
+	for _, event := range events {
+		// process events one by one
+
+		randomized := false
+		var count uint = 1
+		var start string
+		var end string
+
+		if strings.Contains(event, countToken) {
+			//     timespan = time ( "-" / "~" ) time [ "/" ( time / count ) ]
+			ws := strings.Split(event, countToken)
+			eventBase := ws[0]
+			countStr := ws[1]
+			if c, err := strconv.ParseUint(countStr, 10, 32); err != nil || c == 0 {
+				return nil, fmt.Errorf("cannot parse %q: not a valid event interval",
+					event)
+			} else {
+				count = uint(c)
+				event = eventBase
+			}
+
+			// count token is only allowed in timespans, meaning
+			// we're parsing only timespans from now on
+			expectTime = true
+		}
+
+		if special, ok := specialTokens[event]; ok {
+			event = special
+		}
+
+		if strings.Contains(event, randomizedSpanToken) {
+			// timespan uses "~" to indicate that the actual event
+			// time is to be randomized.
+			randomized = true
+			event = strings.Replace(event,
+				randomizedSpanToken,
+				spanToken, 1)
+
+			// randomize token is only allowed in timespans, meaning
+			// we're parsing only timespans from now on
+			expectTime = true
+		}
+
+		// spans
+		//     wdayspan = wday "-" wday
+		//     timespan = time ( "-" / "~" ) time [ "/" ( time / count ) ]
+		start, end = parseSpan(event)
+
+		if isValidTime(start) && isValidTime(end) {
+			// from now on we expect only timespans
+			expectTime = true
+
+			// is it a time?
+			var err error
+			var tstart, tend TimeOfDay
+
+			if tstart, tend, err = parseTimeV2(start, end); err != nil {
+				return nil, err
+			}
+
+			if len(tempScheds) == 0 {
+				// make an every-day schedule if there was none
+				// specified yet
+				tempScheds = append(tempScheds,
+					&Schedule{})
+			}
+
+			// make a schedule for each day that we currently have
+			scheds = append(scheds,
+				makeTimeSchedules(tempScheds, timeScheduleConf{
+					start:     tstart,
+					end:       tend,
+					count:     count,
+					randomize: randomized,
+				})...)
+		} else if isValidWeekdaySpec(start) && isValidWeekdaySpec(end) {
+			// is it a day?
+
+			if expectTime {
+				return nil, fmt.Errorf("cannot parse %q: expected time spec", s)
+			}
+			weSchedule, err := makeWeekdayEvent(start, end)
+			if err != nil {
+				return nil, fmt.Errorf("cannot parse %q: %s", event, err.Error())
+			}
+			tempScheds = append(tempScheds, weSchedule)
+		} else {
+			// no, it's an error
+			return nil, fmt.Errorf("cannot parse %q: not a valid event", event)
```

Comment:
Haven't reviewed the details of the logic above yet. Will leave this for the next round.